app/uk/gov/hmrc/cataloguefrontend/repository/RepositoriesController.scala

```diff
@@ -43,64 +43,75 @@ class RepositoriesController @Inject() (
+      } yield Ok(repositoriesListPage(repositories, teams, RepoListFilter.form.bindFromRequest())
       )}
 
 
   def allServices: Action[AnyContent] =
     Action {
       Redirect(
-        repository.routes.RepositoriesController.allRepositories(repoType = Some(RepoType.Service.asString)))
+        repository.routes.RepositoriesController.allRepositories(repoTypeFilter = Some(RepoType.Service.asString)))
     }
 
   def allLibraries: Action[AnyContent] =
     Action {
-      Redirect(repository.routes.RepositoriesController.allRepositories(repoType = Some(RepoType.Library.asString)))
+      Redirect(repository.routes.RepositoriesController.allRepositories(repoTypeFilter = Some(RepoType.Library.asString)))
     }
 
   def allPrototypes: Action[AnyContent] =
     Action {
-      Redirect(repository.routes.RepositoriesController.allRepositories(repoType = Some(RepoType.Prototype.asString)))
+      Redirect(repository.routes.RepositoriesController.allRepositories(repoTypeFilter = Some(RepoType.Prototype.asString)))
+    }
+}
+
+object RepositoriesController {
+
+  def buildQueryParam(repoTypeFilter: Option[String]): (Option[String], Option[String]) = {
+    repoTypeFilter match {
+      case None                     => (None, None)
+      case Some("Service")          => (Some("Service"), None)
+      case Some("FrontendService")  => (Some("Service"), Some("FrontendService"))
+      case Some("BackendService")   => (Some("Service"), Some("BackendService"))
+      case Some("Library")          => (Some("Library"), None)
+      case Some("Prototype")        => (Some("Prototype"), None)
+      case Some("Other")            => (Some("Other"), None)
+      case _                        => (None, None)
     }
+  }
 }
 
 case class RepoListFilter(
-   name      : Option[String] = None,
-   team      : Option[String] = None,
-   repoType  : Option[String] = None
+                           name: Option[String] = None,
+                           team: Option[String] = None,
+                           repoTypeFilter : Option[String] = None,
                          ) {
   def isEmpty: Boolean =
-    name.isEmpty && team.isEmpty && repoType.isEmpty
+    name.isEmpty && team.isEmpty && repoTypeFilter.isEmpty
 }
 
 object RepoListFilter {
   lazy val form = Form(
   mapping(
-  "name"     -> optional(text).transform[Option[String]](_.filter(_.trim.nonEmpty), identity),
-  "team"     -> optional(text).transform[Option[String]](_.filter(_.trim.nonEmpty), identity),
-  "repoType" -> optional(text).transform[Option[String]](_.filter(_.trim.nonEmpty), identity)
+  "name"            -> optional(text).transform[Option[String]](_.filter(_.trim.nonEmpty), identity),
+  "team"            -> optional(text).transform[Option[String]](_.filter(_.trim.nonEmpty), identity),
+  "repoTypeFilter"  -> optional(text).transform[Option[String]](_.filter(_.trim.nonEmpty), identity)
```

Comment:
Don't think it needs the rename its now RepoListFilter...Filter and doesn't fit in with the name of the other 2 filters.

Reply:
The issue is that repoType is a different thing which is can be Service, Library, Prototype, Other and is a field used for the teams and repos API. 

repoType filter is a combination of repo types and service types (BackendService and FrontendService)

Reply:
The domain is `repoType` and `serviceType`, so we should really be filtering by that. You could use javascript to select a `serviceType` too when selecting the dropdown? The UI shouldn't really influence the RESTful API